src/Mailgun/Mailgun.php

```diff
@@ -36,20 +36,83 @@ class Mailgun
     protected $apiKey;
 
     /**
+     * @var [string => string]
+     */
+    protected $regions = [
+        'si' => 'si.api.mailgun.net',
+        'so' => 'so.api.mailgun.net',
+    ];
+
+    /**
+     * @var [string => RestClient]
+     */
+    protected $regionClients = [];
+
+    /**
      * @param string|null $apiKey
      * @param HttpClient  $httpClient
      * @param string      $apiEndpoint
+     * @param bool        $noRegions
      */
     public function __construct(
         $apiKey = null,
         HttpClient $httpClient = null,
-        $apiEndpoint = 'api.mailgun.net'
+        $apiEndpoint = 'api.mailgun.net',
+        bool $noRegions = false
```

Comment:
We are not exclusive for PHP7. Can't have `bool`.

It is generally bad practice to have feature flags like this.